### DISCUSSION TITLE
Can an Action workflow block codespace creation?

### DISCUSSION THREAD

**Discussion by christianh814** [community/community#121139] [Actions] [answered]
### Select Topic Area

Question

## Summary:

Is there a setting where I can block folks from creating a Codespace on a repo until a GitHub action completes?

## More Info

I have a repo that is a template, I want to create an Action that runs some customization against the repo that is created from this template. This repo/template also has a devcontainer/codespace setting where the user can test the repo in a codespace.

However, the repo won't be ready to test against until that workflow finishes.

Is this possible?

**Comment by grandmas-favorite** [comment, marked as answer]
Hey there @christianh814!

While there is no native feature to do something like that, there are a few options you could use as workarounds. One option is to create the template repo without the `.devcontainer` file in it (or at least not in the correct place of the repo). Then, your "get everything ready" action could have it do all the stuff it is doing now, and then create/move the `.devontainer` files into a state that ia needed to support starting a codespace.

This would NOT block codespace creation until the action completes (aka you could create your own config).

That said, perhaps it makes more sense to have the default README.md indicate that the repo isn't ready for codespace creation yet, and have the Action update the README.md file when it runs? 

There's also the `first-run-notice.txt file` that VS code displays in the terminal session that could convey state information. It can be put into the right location from the `.devcontainer` directory by adding this to the `devcontainer.json` file:
`"postCreateCommand": "sudo cp .devcontainer/welcome.txt /usr/local/etc/vscode-dev-containers/first-run-notice.txt"`
 
I hope this helps get you started!
